src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -165,6 +166,10 @@ def self.client
     @client ||= create_client
   end
 
+  def self.client=(new_client)
+    @client = new_client
+  end
```

Comment:
@camlow325 I added this setter method, which... is kind of crappy because I wouldn't want it to be called from anywhere else, but, maintains the abstraction of the class variable so that knowledge doesn't leak out into the other code.  This is kind of gross, but I think the other way would be gross in a different way.  If you have a strong opinion I can flip it, though.